packages/horizon/ignition/modules/periphery/GraphToken.ts

```diff
@@ -0,0 +1,33 @@
+import { buildModule } from '@nomicfoundation/hardhat-ignition/modules'
+import { deployWithGraphProxy } from '../proxy/GraphProxy'
+
+import GraphTokenGatewayModule from '../periphery/GraphTokenGateway'
+import RewardsManagerModule from '../periphery/RewardsManager'
+
+import GraphTokenArtifact from '@graphprotocol/contracts/build/contracts/contracts/l2/token/L2GraphToken.sol/L2GraphToken.json'
+
+// TODO: Ownership transfer is a two step process, the new owner needs to accept it by calling acceptOwnership
+export default buildModule('GraphToken', (m) => {
+  const { RewardsManager } = m.useModule(RewardsManagerModule)
+  const { GraphTokenGateway } = m.useModule(GraphTokenGatewayModule)
+
+  const deployer = m.getAccount(0)
+  const governor = m.getParameter('governor')
+  const initialSupply = m.getParameter('initialSupply')
+
+  const { instance: GraphToken } = deployWithGraphProxy(m, {
+    name: 'GraphToken',
+    artifact: GraphTokenArtifact,
+    args: [deployer],
+  })
+
+  // TODO: move this mint to a testnet only module
```

Comment:
How have you been managing testnet/mainnet deployment differences so far?

Reply:
I was actually wondering that. So far I have not been managing it at all. Everything so far assumes it's a testnet deployment, I'm curious what patterns you've seen for this? 

I can think of two approaches:
1. Add testnet/mainnet modules with specific config/steps for each environment. Those modules would import the main Horizon module, deploy everything in there then add on top of that.
2. Deploy with a script instead of CLI and have any logic in there to deal with differences.

Reply:
Another idea that we had explored was allowing the users to create "chain specific" modules, that would behave different depending on the chainId. Imagine something like this:

```
const mod = chainSpecificModule("Mod", 
{
   1: (m) => {
      // mainnet logic
   },
   31337: (m) => {
      // hardhat network logic
   }
})
```

(ignore the horrible syntax)

This could work, especially if all you do are operations inside of those modules. If you need to export contracts, maybe we'd have to force you to export the same contract futures or something like that, so that it plays nicely with typescript.

Do you normally do ops there?

Reply:
Another pattern would be something like having your `main` module, and then for testnets you have `testnet` modules, which `useModule(main)`, and run those ops. This feels closer to extending a class for testing purposes.

The disadvantage there is that is may force you to structure `main` in a certain way so that you can do those ops.

Reply:
That last pattern you described is what I had in mind. Chain specific modules would be really helpful though, there are certain operations for which I think it makes a lot of sense (for example, minting initial supply of tokens on a testnet)